framework/src/media/OutputHandler.cpp

```diff
@@ -210,6 +203,31 @@ void OutputHandler::destroyWorker()
 	}
 }
 
+bool OutputHandler::writeToSource(size_t size)
+{
+	auto buf = new unsigned char[size];
+	bool ret = true;
+	size_t wlen = 0;
+	while (wlen < size) {
+		size_t temp = mBufferReader->read(buf + wlen, size - wlen);
+		if (temp == 0) {
+			// Can not read data, error occurred.
+			meddbg("StreamBufferReader::read failed!\n");
+			break;
+		}
+		wlen += temp;
+	}
+
+	if (mOutputDataSource->write(buf, wlen) < 0) {
```

Comment:
What happens when wlen is zero?
Subclasses of outputdatasource may not handle this case properly.

Reply:
I fixed like this
https://github.com/Samsung/TizenRT/pull/2333
Would you review this PR?

Reply:
Please, double-check if returning 0 means success or failure.
In #2333, 0 is returned when the size is 0.

Reply:
Now if `mBufferReader->read` failed, `OutputHandler::writeToSource` returns false.
It can handle wlen == 0 case

Reply:
wlen is the number of readed data from mBufferReader.
Usually, wlen and size is same. But write(buf, wlen) seems better for the worst case

Reply:
Let's think about more.
In this case, read size is not same as data size in buffer, this means there's chance that data remained in ring buffer.
Now we are flushing, in this case we should clean ring buffer.
So more logic needed.
```
size_t wlen = mBufferReader->read(buf, size);
if (wlen != size) {
     delete buf;
     return false; //we'd better do nothing than write invalid data
}

....

if (stream->mIsFlushing) { // check workerMain
   do_something;
    clean_rbs
}
```

Reply:
I also agree with @Taejun-Kwon.

Reply:
I think the error case should include '0' case as well as follows:
if (mOutputDataSource->write(buf, wlen) <= 0) {